(Fixes #236) Add support for exalead

bhaveshAn opened the pull request:
<!-- Add the issue number that is fixed by this PR (In the form Fixes #123) -->
Fixes #236 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [ ] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added support for Exalead

cclauss commented:
Reading code in the GitHub web UI is a common practice.  Code wider than 127 characters requires scrolling left and right in that UI which is highly sub-optimal for the reader.

I would rather see you wrap these lines or that you tell flake8 to excludes the file app/templates/index.html or the directory app/templates/  See the flake8 docs on how to do this.

bhaveshAn commented:
>  tell flake8 to excludes the file app/templates/index.html or the directory app/templates/ See the flake8 docs on how to do this.

Flake8 is used for python files only and not for HTML files or other files and this build is failing due to [this](https://github.com/fossasia/query-server/pull/240/files#diff-73a1e08ff08aa50971c8321d19055e5eR43) and discussion is also for the mentioned line [here](https://github.com/fossasia/query-server/pull/240/files#diff-73a1e08ff08aa50971c8321d19055e5eR43)

Thanks for review.

bhaveshAn commented:
Further rebase will make the previous 2 upstream commits into mine. Hence no need to rebase further.
@cclauss @gabru-md please review.

bhaveshAn commented:
Further rebase will make the previous 2 upstream commits into mine. Hence no need to rebase further.
@cclauss @gabru-md please review. [Example video at Youtube](https://www.youtube.com/watch?v=D1YBrNAiDZc)

bhaveshAn commented:
> And please don't mix so many into a single PR/issue. Ideally, each of the search engine implementation should be a separate issue-PR pair so that implementation/requirements can be discussed.

I will do this from next time. 

@niranjan94 please review :)

niranjan94 commented:
@bhaveshAn can we have a test deployment please ? :)

bhaveshAn commented:
@niranjan94 can i use ngrok ?

niranjan94 commented:
@bhaveshAn ngrok would require you to have a local instance up and running. Hence it isn't always reliable as many times we'll end up with dead links and late reviews. Please use heroku. Easy and free.

bhaveshAn commented:
Did but can you tell me what to do for bower ?

bhaveshAn commented:
@niranjan94 For now it looks https://ancient-wave-72336.herokuapp.com/
can you help me? please

niranjan94 commented:
@bhaveshAn how did you deploy to heroku ? Bower should automatically get downloaded as long as the nodejs build pack is present.

bhaveshAn commented:
actually i ran ``heroku buildpacks:set heroku/python``` and then push

niranjan94 commented:
https://github.com/fossasia/query-server just use the one-click deployment button on README

bhaveshAn commented:
> @bhaveshAn how did you deploy to heroku ? Bower should automatically get downloaded as long as the nodejs build pack is present.

After `git push heroku bhavesh-yandex:master` it didn't installed bower

AnshulMalik commented:
Hey @bhaveshAn, You also need to have the node buildpack.

bhaveshAn commented:
Did that alongwith python buildpack. Didn' succeed.
@AnshulMalik Can you please review the PR.

bhaveshAn commented:
@gabru-md @S2606 please review.
[Youtube Video](https://www.youtube.com/watch?v=zGdUT9f3vvE)

niranjan94 commented:
@bhaveshAn please provide a deployment here

bhaveshAn commented:
@niranjan94 
https://shrouded-reef-37326.herokuapp.com/ This heroku deployment is working for Exalead and Teoma and not for Lycos.
While all the three running locally and at http://0ebf3762.ngrok.io/

niranjan94 commented:
Please debug the heroku deployment and fix lycos. We need everything to
work on heroku..

On Sun, Oct 15, 2017, 8:47 PM Bhavesh Anand <notifications@github.com>
wrote:

> @niranjan94 <https://github.com/niranjan94>
> https://shrouded-reef-37326.herokuapp.com/ This heroku deployment is
> working for Exalead and Teoma and not for Lycos.
> While all the three running locally and at http://0ebf3762.ngrok.io/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/fossasia/query-server/pull/240#issuecomment-336773396>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACSwFO6nHAlnYOSY31fVQwydpL-1OCJzks5sstHYgaJpZM4PxpyM>
> .
>

bhaveshAn commented:
@niranjan94 Sure. I will let you know after debugging/fixing it.
Apologies, Did down thumbs emoji by mistake in previous comment. Hope you don't mind :disappointed:

bhaveshAn commented:
As Lycos wasn't able to run on heroku, so removing it. Will create separate issue and PR.
Heroku deployment for Exalead and Teoma is at https://limitless-mesa-39171.herokuapp.com/
@niranjan94 @mariobehling Please review. Thanks !

AnshulMalik commented:
Hey @bhaveshAn , I tried to list 40 results of teoma, but looks like they were duplicates, can you look into it please?

mariobehling commented:
Please take out Teoma as its results come from Ask.com. Also, resolve merge conflicts. Cheers.

mariobehling commented:
Please take out your Teoma references and rebase. Cheers.

bhaveshAn commented:
@mariobehling Please review. Removed Teoma from the files and squashed the commits. Thanks !

bhaveshAn commented:
@mariobehling Please review. Removed Teoma from the files and squashed the commits. Thanks !
@niranjan94 please review . Providing the heroku deployment at https://powerful-shelf-77291.herokuapp.com/

bhaveshAn commented:
Exalead showing all the 100 results.
@vaibhavsingh97 @cclauss @shashank-sharma please review. Providing the heroku deployment at https://powerful-shelf-77291.herokuapp.com/

cclauss commented on app/scrapers/exalead.py:
```diff
@@ -0,0 +1,25 @@
+                'title': a.getText(),
+                'link': a.get('href')
+                })
+        print('parsed' + str(urls))
```
OPTIONAL: Would it make sense to change this line to __print('Exalead parsed: ' + str(urls))__ so that it is easier to know that the results came from the expected scraper?

gabru-md replied:
@cclauss i guess it doesn't matter. As it will be on the debuggers side where the print command would work and the scrapper can be known using the api comand whenever a new search is done as it is given in the query parameters.
Cool ?

cclauss replied:
If it does not matter then please explain why #271 was so difficult for us to discover?

gabru-md replied:
That was just my opinion @cclauss.
After going through #271 i believe that you are right in saying so.
I apologise for the mistake. :)

gabru-md replied:
Agter going througj the code i believe that i might have found the possible fix for #271 as well. I will try to work it out and send a PR  as soon as possible

cclauss replied:
@bhaveshAn It is optional but your thoughts on this one?

bhaveshAn replied:
Now check. :smiley:

sudheesh001 commented on app/scrapers/exalead.py:
```diff
@@ -0,0 +1,25 @@
+
+
+class Exalead(Scraper):
+    """Scrapper class for Exalead"""
```
Typo `Scrapper` => `Scraper`

bhaveshAn replied:
@sudheesh001 could you please review again ??
